torch/csrc/jit/script/module.cpp

```diff
@@ -213,10 +223,40 @@ static FunctionSchema sliceFirst(const FunctionSchema& schema) {
   return schema.cloneWithArguments(std::move(sliced));
 }
 
-Method::Method(Module* owner, Function* first_class_function)
+Method::Method(
+    Module* owner,
+    const std::shared_ptr<Function>& first_class_function)
     : owner_(owner), schema_(sliceFirst(first_class_function->getSchema())) {
-  std::tie(function_, initial_ivalues_) =
-      owner->lower_first_class_method(first_class_function);
+  if (experimental_run_as_first_class) {
+    function_ = first_class_function;
+    // initial_ivalues_ left blank
+  } else {
+    std::tie(function_, initial_ivalues_) =
+        owner->lower_first_class_method(first_class_function.get());
+  }
+}
+
+void Method::run(Stack& stack) {
+  if (experimental_run_as_first_class) {
+    stack.insert(stack.begin(), owner().module_object());
+  }
+  for (const auto& input : initial_ivalues_) {
+    push(stack, input.value());
+  }
+  function_->run(stack);
+}
+
+IValue Method::operator()(std::vector<IValue> stack, const Kwargs& kwargs) {
+  getSchema().checkAndNormalizeInputs(stack, kwargs);
+  if (experimental_run_as_first_class) {
```

Comment:
nitpick: this logic (pushing module and parameters onto a stack) seems to be duplicated across `run` and `()`. Maybe we should consider refactoring it into a helper, otherwise we might forget to update one of the places eventually?